.github/workflows/android_builds.yml

```diff
@@ -6,7 +6,7 @@ on:
 env:
   # Used for the cache key. Add version suffix to force clean build.
   GODOT_BASE_BRANCH: master
-  SCONSFLAGS: verbose=yes warnings=extra werror=yes debug_symbols=no module_text_server_fb_enabled=yes
+  SCONSFLAGS: verbose=yes warnings=extra werror=yes debug_symbols=no module_text_server_fb_enabled=yes use_strict_ref_checks=yes
```

Comment:
What do you think of using `check_reference_comparison=yes`?

The current naming is a bit cumbersome and ref is a shorthand.

Reply:
Maybe `check_refcounted_comparison` or `check_ref_comparsion`.

Reply:
Note that `ref` here refers to the `Ref` container, not `RefCounted` so `ref` is appropriate here IMO

What about `no_ref_null_compare`? Or as an inverted one defaulting to `true`, `ref_null_compare` or `ref_nullptr_compare`?

Reply:
`ref_null_compare=yes` Sounds good to me.